lib/improver/tests/utilities/test_cube_manipulation.py

```diff
@@ -527,6 +545,710 @@ def test_cubelist_different_var_names(self):
+        with self.assertRaisesRegexp(ValueError, msg):
+            equalise_cube_coords(cubes)
+
+    def test_model_id_with_realization_in_cube(self):
+        """Test if model_id is an unmatched coordinate, a cube has a
+        realization coordinate and the cube being inspected has a realization
+        coordinate."""
+        cube1 = self.cube.copy()
+        cube2 = self.cube.copy()[0]
+        cube2.remove_coord("realization")
+        model_id_coord = DimCoord(
+            np.array([100*1], np.int), long_name='model_id')
+        cube1.add_aux_coord(model_id_coord)
+        cube1 = iris.util.new_axis(cube1, "model_id")
+        cubes = iris.cube.CubeList([cube1, cube2])
+        result = equalise_cube_coords(cubes)
+        self.assertIsInstance(result, iris.cube.CubeList)
+        self.assertEqual(len(result), 4)
+        self.assertTrue(result[0].coords("realization"))
+        self.assertFalse(result[3].coords("realization"))
+        self.assertTrue(result[0].coords("model_id"))
+
+    def test_model_id_with_realization_not_in_cube(self):
+        """Test if model_id is an unmatched coordinate, a cube has a
+        realization coordinate and the cube being inspected does not have a
+        realization coordinate."""
+        cube1 = self.cube.copy()
+        cube2 = self.cube.copy()
+        cube1.remove_coord("realization")
+        model_id_coord = DimCoord(
+            np.array([100*1], np.int), long_name='model_id')
+        cube2.add_aux_coord(model_id_coord)
+        cube2 = iris.util.new_axis(cube2, "model_id")
+        cubes = iris.cube.CubeList([cube1, cube2])
+        result = equalise_cube_coords(cubes)
+        self.assertIsInstance(result, iris.cube.CubeList)
+        self.assertEqual(len(result), 4)
+        self.assertFalse(result[0].coords("realization"))
+        self.assertTrue(result[1].coords("realization"))
+        self.assertTrue(result[1].coords("model_id"))
+
+
+class Test_compare_attributes(IrisTest):
+    """Test the compare_attributes utility."""
+
+    def setUp(self):
+        """Use temperature cube to test with."""
+        self.cube = set_up_temperature_cube()
+        self.cube_ukv = self.cube.extract(iris.Constraint(realization=1))
+        self.cube_ukv.remove_coord('realization')
+        self.cube_ukv.attributes.update({'grid_id': 'ukvx_standard_v1'})
+        self.cube_ukv.attributes.update({'title':
+                                         'Operational UKV Model Forecast'})
+        self.cube.attributes.update({'grid_id': 'enukx_standard_v1'})
+        self.cube.attributes.update({'title':
+                                     'Operational Mogreps UK Model Forecast'})
+
+    def test_basic(self):
+        """Test that the utility returns a list and have no differences."""
+        cube1 = self.cube.copy()
+        cube2 = self.cube.copy()
+        cubelist = iris.cube.CubeList([cube1, cube2])
+        result1 = compare_attributes(cubelist)
```

Comment:
Why result1 and not just result?

Reply:
Because compare_attributes used to return two lists, a list of all the attributes that matched and the unmatching attributes; because we never made use of the first list I removed it to get it pass codacy. I'll change it to result